models/ConfigureForm.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+namespace humhub\modules\socialshare\models;
+
+use Yii;
+use yii\base\Model;
+
+class ConfigureForm extends Model
+{
+    public $facebook_enabled;
+    public $twitter_enabled;
+    public $linkedin_enabled;
+    public $line_enabled;
+    public $bluesky_enabled;
+    public $custom_facebook_url;
+    public $custom_twitter_url;
+    public $custom_linkedin_url;
+    public $custom_line_url;
+    public $custom_bluesky_url;
+
+    /**
+     * @inheritdoc
+     */
+    public function rules()
+    {
+        return [
+            [['facebook_enabled', 'twitter_enabled', 'linkedin_enabled', 'line_enabled', 'bluesky_enabled'], 'boolean'],
+            [['custom_facebook_url', 'custom_twitter_url', 'custom_linkedin_url', 'custom_line_url', 'custom_bluesky_url'], 'string'],
+        ];
+    }
+
+    /**
+     * Loads settings from the module's configuration
+     */
+    public function loadSettings()
```

Comment:
I think better to move the code to the existing method `init()` instead of call this `$settings->loadSettings();` after each `new ConfigureForm()`.

Reply:
Good point, I've tested and it works rather well!